kore/src/Kore/Builtin/Map.hs

```diff
@@ -499,10 +499,9 @@ internalize ::
     TermLike variable ->
     TermLike variable
 internalize tools termLike
-    | fromMaybe False (isMapSort tools sort')
```

Comment:
Are we just removing this check because it is redundant? Or is there another reason?

Reply:
This is the check that I asked you about the other day, the one that Evan removed in [his PR](https://github.com/kframework/kore/pull/2440) that I then had to revert. You suggested I leave it in even if it doesn't solve the Issue linked to this PR.

Reply:
Oh, that is not the check I thought you were removing. It **is** redundant, and I'm glad to be rid of it, but it doesn't have anything to do with this issue in particular.

Reply:
So, is it fine to leave it in this PR or should I open a new PR?